packager/media/base/aes_encryptor.cc

```diff
@@ -129,17 +129,40 @@ bool AesCbcEncryptor::CryptInternal(const uint8_t* plaintext,
                                     size_t plaintext_size,
                                     uint8_t* ciphertext,
                                     size_t* ciphertext_size) {
-  const size_t residual_block_size = plaintext_size % AES_BLOCK_SIZE;
+
   const size_t num_padding_bytes = NumPaddingBytes(plaintext_size);
-  // mbedtls requires a buffer large enough for one extra block.
   const size_t required_ciphertext_size =
       plaintext_size + num_padding_bytes + AES_BLOCK_SIZE;
+
   if (*ciphertext_size < required_ciphertext_size) {
+    VLOG(1) << "Expected output size of at least "
+               << required_ciphertext_size << " bytes. Using teporary buffer.";
+
+    std::unique_ptr<uint8_t[]> ciphertext_buffer(new uint8_t[required_ciphertext_size]);
```

Comment:
I don't recall having to do this anywhere else.  I think it is expected that we will allocate the right size buffer before calling this method, so that we aren't churning the allocator unnecessarily.  We should control all call sites, and any encryption failure should be fixed at the caller.

At least, this is how we've done it so far.  Please make your case if you disagree.

Reply:
@zdanek thoughts?

Reply:
I reverted this to the previous behavior, to the extent we need to deal with callers passing in the wrong size buffer we can do that in a separate pull request, keeping this one focused on just changes needed to move to CMake.